Carting/src/WebApi/ConfigureServices.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Reflection;
+using Carting.WebApi.Application.Common.Behaviours;
+using Carting.WebApi.Application.Common.Interfaces;
+using Carting.WebApi.Filters;
+using Carting.WebApi.Infrastructure.Persistence;
+using Carting.WebApi.Infrastructure.Services;
+using FluentValidation;
+using FluentValidation.AspNetCore;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+public static class ConfigureServices
+{
+    public static IServiceCollection AddWebUIServices(this IServiceCollection services)
+    {
+        services.AddHttpContextAccessor();
+
+        services.AddHealthChecks();
+
+        services.AddControllersWithViews(options =>
+            options.Filters.Add<ApiExceptionFilterAttribute>())
+                .AddFluentValidation(x => x.AutomaticValidationEnabled = false);
+
+        // Customise default API behaviour
+        services.Configure<ApiBehaviorOptions>(options =>
+            options.SuppressModelStateInvalidFilter = true);
+
+        services.AddOpenApiDocument(configure =>
+        {
+            configure.Title = "Carting API";
+        });
+
+        return services;
+    }
+
+    public static IServiceCollection AddInfrastructureServices(this IServiceCollection services, IConfiguration configuration)
+    {
+        services.AddScoped<ICartingDbContext, CartingDbContext>();
+        services.AddTransient<IDateTime, DateTimeService>();
```

Comment:
That's a great question. We often have lengthy conversions about the service lifetimes and what strategy we should follow. So far, I've yet to come to a conclusion which is better:

- Make everyting singleton, unless you have a good reason to use other lifetime settings. For sure, this is performing better, but also have a greater chance to "turn things to something bad"
- Make everything transient, unless you have a good reason to use other lifetime settings. This is less risky but of course will result in a lot of unnecessary object creation. But the reason I hear a lot, is that "if you want to write such performing code where these differences count, then you shouldn't have chosen C# language any way".

I'd be interested to hear your opinion about it, and what strategies you typically follow here.

Reply:
I prefer KISS:
* If service is clearly can be static like DateTimeService which is a good example, since this is a wrapper over system clock,
just use it.
* If object requires scope associated data when I use scoped. 

Now that I think I don't remember any case where Transient objects was used.